Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -95,13 +95,12 @@ void UPSHUDComponent::SavePoints(EEndGameState EndGameState)
 // Listening game states changes events 
 void UPSHUDComponent::OnGameStateChanged(ECurrentGameState CurrentGameState)
 {
-	CurrentGameStateInternal = CurrentGameState;
 	if (!ensureMsgf(ProgressionMenuWidgetInternal, TEXT("ASSERT: [%i] %hs:\n'ProgressionMenuWidgetInternal' is null!"), __LINE__, __FUNCTION__))
 	{
 		return;
 	}
 
-	switch (CurrentGameState)
+	switch (AMyGameStateBase::GetCurrentGameState())
```

Comment:
You already in the game state function, use `CurrentGameState` function's argument instead of `AMyGameStateBase::GetCurrentGameState()`

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/f1ad376286c5bb905b2fa7568129c6b1ad74dd81